js/initcontext.go

```diff
@@ -288,13 +71,68 @@ func readFile(fileSystem afero.Fs, filename string) (data []byte, err error) {
 }
 
 // allowOnlyOpenedFiles enables seen only files
-func (i *InitContext) allowOnlyOpenedFiles() {
-	fs := i.filesystems["file"]
-
+func allowOnlyOpenedFiles(fs afero.Fs) {
 	alreadyOpenedFS, ok := fs.(fsext.OnlyCachedEnabler)
 	if !ok {
 		return
 	}
 
 	alreadyOpenedFS.AllowOnlyCached()
 }
+
+type requireImpl struct {
+	vu      modules.VU
+	modules *moduleSystem
+	pwd     *url.URL
+}
+
+func (r *requireImpl) require(specifier string) (*goja.Object, error) {
+	// TODO remove this in the future when we address https://github.com/grafana/k6/issues/2674
+	// This is currently needed as each time require is called we need to record it's new pwd
+	// to be used if a require *or* open is used within the file as they are relative to the
+	// latest call to require.
+	// This is *not* the actual require behaviour defined in commonJS as it is actually always relative
+	// to the file it is in. This is unlikely to be an issue but this code is here to keep backwards
+	// compatibility *for now*.
+	// With native ESM this won't even be possible as `require` might not be called - instead an import
+	// might be used in which case we won't be able to be doing this hack. In that case we either will
+	// need some goja specific helper or to use stack traces as goja_nodejs does.
+	currentPWD := r.pwd
+	if specifier != "k6" && !strings.HasPrefix(specifier, "k6/") {
+		defer func() {
+			r.pwd = currentPWD
```

Comment:
`currentPWD` is redundant, so maybe just `currWD`. I would also rename all uses of `pwd` to `cwd`, but that would touch many places (`loader`, tests, etc.), so let's leave that as is.

Reply:
`currentPWD` is what is used for the current require the one that is being called at the moment.
`r.pwd` is set to the value it will need when `require` (or `open`) is called *within* the thing that we will require with this one.

Reply:
I get what it's used for :)

My point is that `pwd` really means `cwd`, and "current current working directory" is redundant. But it's not a big deal.